client/src/glossary/GlossaryMenuController.js

```diff
@@ -0,0 +1,156 @@
+      const glossary_item_key = target.dataset.ibttGlossaryKey;
+      const glossary_item = glossaryEntryStore.lookup(glossary_item_key);
+
+      this.setGlossaryItem(glossary_item.id);
+      this.setList(false);
+
+      this.toggleGlossary(true);
+      document.querySelector(`.glossary-sb__search-bar > input`).focus();
+    };
+
+    closeSidebar = (e) => {
+      const menu_node = this.menu_ref.current;
+      if (
+        this.state.showGlossary &&
+        menu_node &&
+        !menu_node.contains(e.target)
+      ) {
+        this.setState({ showGlossary: false });
+      }
+    };
+
+    handleWindowClick = (e) => {
+      const target = e.target.closest("[data-toggle=glossary_sidebar]");
+      target ? this.itemClick(target) : this.closeSidebar(e);
+    };
+
+    handleWindowKeyDown = (e) => {
+      if (e.code === "Enter") {
+        this.handleWindowClick(e);
+      }
+    };
+
+    componentDidMount() {
+      window.addEventListener("click", this.handleWindowClick, {
+        capture: true,
+      });
+      window.addEventListener("keydown", this.handleWindowKeyDown, {
+        capture: true,
+      });
+    }
+
+    componentWillUnmount() {
+      window.removeEventListener("click", this.handleWindowClick, {
+        capture: true,
+      });
+      window.removeEventListener("keydown", this.handleWindowKeyDown, {
+        capture: true,
+      });
+    }
+
+    toggleGlossary(value) {
+      this.setState({
+        showGlossary: value,
+      });
+    }
+
+    setGlossaryItem(key) {
+      this.setState({
+        glossaryItem: glossaryEntryStore.lookup(key),
+        showList: false,
+      });
+    }
+
+    setList(value) {
+      this.setState({
+        showList: value,
+      });
+    }
+
+    setQuery(query) {
+      this.setState({ query: query });
+    }
+
+    setResults = (childData) => {
+      const test = _.map(childData, (data) => ({
+        id: data.glossary_data.id,
+        title: data.glossary_data.title,
+        translation: data.glossary_data.translation,
+        raw_definition: data.glossary_data.raw_defintion,
+        get_compiled_definition: data.glossary_data.get_compiled_definition,
+      }));
+
+      this.setState({
+        results: test,
+      });
+      this.setList(true);
+    };
+
+    render() {
+      const currentPage = this.props.location.pathname;
+
+      if (ROUTES_WITHOUT_GLOSSARY[currentPage]) return null;
+
+      return (
+        <div ref={this.menu_ref}>
+          <GlossaryMenu
+            show={this.state.showGlossary}
+            toggle={(value) => this.toggleGlossary(value)}
+            item={this.state.glossaryItem}
+            setGlossaryItem={(key) => this.setGlossaryItem(key)}
+            showList={this.state.showList}
+            setList={(value) => this.setList(value)}
+            setResults={(data) => this.setResults(data)}
+            setQuery={(query) => this.setQuery(query)}
+            results={get_glossary_items_by_letter(this.state.results)}
```

Comment:
Grouping results by letter should be `GlossaryMenu`'s responsibility (or this prop should have a name like `resultsByLetter` or something, but that's more clumsy).